src/server/components/notebook-actions-menu.jsx

```diff
@@ -57,7 +57,7 @@ export default class NotebookActionsMenu extends React.Component {
       let filename;
```

Comment:
lol @openjck you should not have drawn attention to this method, now i'm going to ask you to refactor it ;-)

- we're trying to move toward async/await rather than `.then` -- can you convert this to an async method and await `selectFileAndFormatMetadata`?
- these funny `catch (err) { throw err; }` blocks keep showing up in our code. please be proactive in destroying them wherever you find the :-)
- but in this case, it looks like we might refactor this function using `await` an differently structured try/catch to both simplify this code AND to get rid of the `// eslint-disable-line prefer-destructuring` flag. how about somthing like:

```
try{
// try to do whatever we'd want id the file does exist...
const formdata = await selectFileAndFormatMetadata(...)
...etc code...
} catch {
// do whatever we need to do if the file does not exist
...etc code...
}
```

it looks like that is the actual logical structure here, and that we can get rid of the if/else below also

(cc @hamilton who knows this part of the code better than i do)

Reply:
Sure! How do you feel about doing this in a follow-up? This PR has gotten long in the tooth so I'd prefer not to block on this.